pkg/vsphere/tasks/waiter.go

```diff
@@ -82,3 +87,29 @@ func WaitForResult(ctx context.Context, f func(context.Context) (ResultWaiter, e
 	}
 	return info, nil
 }
+
+func WaitAndRetryForResult(ctx context.Context, f func(context.Context) (ResultWaiter, error)) (*types.TaskInfo, error) {
```

Comment:
Heeded your advice :) there is now no longer a collision limit. Only an infinite loop and a possible CDE. we should note though, that if no deadline is set we now have an infinite loop.